osquery/tables/system/darwin/firewall.cpp

```diff
@@ -43,6 +44,58 @@ const std::map<std::string, std::string> kTopLevelStringKeys{
     {"version", "version"},
 };
 
+Status parseApplicationAliasData(const std::string& data, std::string& result) {
+  std::string decoded_data = base64Decode(data);
+  if (decoded_data.empty()) {
+    return Status(1, "Failed to base64 decode data");
+  }
+
+  CFDataRef resourceData = CFDataCreate(
+      nullptr,
+      static_cast<const UInt8*>(static_cast<const void*>(decoded_data.c_str())),
+      decoded_data.length());
+  if (resourceData == nullptr) {
+    return Status(1, "Failed to allocate resource data");
+  }
+
+  auto alias = (CFDataRef)CFPropertyListCreateWithData(kCFAllocatorDefault,
+                                                       resourceData,
+                                                       kCFPropertyListImmutable,
+                                                       nullptr,
+                                                       nullptr);
+  CFRelease(resourceData);
+  if (alias == nullptr) {
+    return Status(1, "Failed to allocate alias data");
+  }
+
+  auto bookmark =
+      CFURLCreateBookmarkDataFromAliasRecord(kCFAllocatorDefault, alias);
+  CFRelease(alias);
+  if (bookmark == nullptr) {
+    return Status(1, "Alias data is not a bookmark");
+  }
+
+  auto url = CFURLCreateByResolvingBookmarkData(
+      kCFAllocatorDefault, bookmark, 0, nullptr, nullptr, nullptr, nullptr);
+  CFRelease(bookmark);
+  if (url == nullptr) {
+    return Status(1, "Alias data is not a URL bookmark");
+  }
+
+  // Get the URL-formatted path.
+  result = stringFromCFString(CFURLCreateStringByReplacingPercentEscapes(
```

Comment:
I think you will leak memory here.
```
CFURLCreateStringByReplacingPercentEscapes
```
Will allocate and return a temporary to `stringFromCFString` but that allocation will never be released. You'll have to name it, then release it.
```cpp
auto replaced = CFURLCreateStringByReplacingPercentEscapes(kCFAllocatorDefault, CFURLGetString(url), CFSTR(""));
CFRelease(url);
if (replaced == nullptr) {
  return ...
}
result = stringFromCFString(replaced);
CFRelease(replaced);
...
```

Reply:
Ah, good catch.  I'll make that change and re-submit. Thx.